### ISSUE TITLE
Ontology related data

### ISSUE THREAD

**Issue by davmlaw** [SACGF/variantgrid#423]
It would be good to be able to look up all of our data that has been linked to an ontology, eg if you're interested in AML, see all classifications / patients / samples etc

* Write an ontology page that has autocompletes for each term, and jumps to the ontology pages
* On the bottom of ontology page, do a "related data" section

**Comment by TheMadBug** [comment]
Rather than a dedicated page with autocomplete, could the general search just become a plain text ontology search, and if there are multiple matches it goes into search results?

**Comment by davmlaw** [comment]
Yeah, agree, it would have been a pretty lame page with 1 autocomplete on it.

The trouble is it wrecks the "jump to gene symbol" search functionality, even if you exclude HGNC as MONDO has lots of exact gene names, and eg "RUNX1" matches "....predisposition syndrome associated with RUNX1" - so I made search skip ontology if the string matches a gene symbol.

---

**Future work**

I currently only show classifications / patients etc if it's an exact match for the ontology, not taking into account the hierarchy, ie "cancer" won't show AML 

IMHO our ontology page is better than the external one, even before I add all this related data etc, might be worth thinking about switching links (eg gene page) to our internal one. I think Sarah asked to change it to external, might have to debate this

**Comment by davmlaw** [comment]
It was a bit easy to get lost in a search as eg "kabuki" returned 5 Ontology results, so I added a summary eg the one things are classified against etc has "Condition for 1 classification, 1 patient with 2 samples"

This was originally there for Variant only, and was disabled in Shariant as it displayed ClinVar info. I've now just disabled the Clinvar in Shariant, settings are:

```
SEARCH_SUMMARY = True  # Little tips that show up on search
SEARCH_SUMMARY_VARIANT_SHOW_CLINVAR = False
```

**Comment by davmlaw** [comment]
Should run through Shariant test, can disable via settings if Emma doesn't like it

**Comment by TheMadBug** [comment]
@davmlaw I've added a few enhancements after looking at it:

Search results of any type will now be limited to 50 and an error message will appear at the top saying how many results there were. I did this because if you search for "syndrome" you get over 7,000 results.

Search results are now sorted by ontology type and then name. Sadly this does give priority to HP over MONDO but I feel that is better tackled with another rewrite of how the search framework is handled.

In the search instructions I have linked off to an external MONDO search. This is because we still get caught out in edge cases (e.g. if the official term end with "iv" searching for "4" wont match that). I've gotten around this before by using an external search API for condition text matching, but I'd rather not introduce that point of failure and delay into the general search.

Rather than have a toggle for each group of relationships on the ontology page, I've put a single toggle for all of them. If the number of relationships is 30 or under, it will be expanded by default, otherwise it will be collapsed by default.

**Comment by davmlaw** [comment]
Cool, thanks all really good stuff to do. I've also added a v2 issue for more work at #426

**Comment by EmmaTudini** [comment]
Testing:
- Logged in as non-admin assigned to VCGS Unit 1
1. Searched for "breast cancer" and "melanoma"
2. Searched for HP:0011524, MONDO:0012345, diabetes mellitus, noninsulin-dependent, 1 and OMIM:615916. 
3. Clicked on HGNC links to TGM5 in MONDO:0012345
4. Searched for HGNC:MLH1 and HGNC:7127
5. Went to https://test.shariant.org.au/ontology/term/HP_0004308 and counted the number of gene/ontology relationships. 
6. Went to https://test.shariant.org.au/ontology/term/HP_0000118

Expected output: 1. Should be shown all ontology terms with "breast cancer" and "melanoma"
2. Should show normal ontology page with classifications related underneath in created date order (same as classification listing)
3. Should be taken to HGNC page with no classifications listed
4. As above
5. Says 6 ontology/gene relationships so should display 6. Should be auto-expanded. 
6. SHould be auto-not expanded

Actual output: 
1. Passed 
Note: @TheMadBug a bunch of obsolete mondo terms also show in this search. I wonder if we should filter them out or at least have them at the bottom?
Also, it does cut off all results after 50. Should this allow for a toggle to expand? Or if not, should we show the MONDO terms first (think it's currently alphabetical with HPO first)
2. Passed
3. Passed
4. Passed
5. **Failed** - not sure how these are counted but there seem to be more than 6. Same for HP:0011675. I think what might be happening is that if a gene relationship exists, then it shows the number of relationships - see https://test.shariant.org.au/ontology/term/MONDO_0010466. But if no gene relationships exists, then it counts all of the relationships. See - https://test.shariant.org.au/ontology/term/MONDO_0005217
6. Passed

Note: In the search help - what does "or for a dedicated MONDO search OLS Ontology Search" mean?

**Comment by TheMadBug** [comment]
I've now excluded terms with the word 'obsolete' in them unless you specifically include the word 'obsolete' in your search.

Re the sort order
> Search results are now sorted by ontology type and then name. Sadly this does give priority to HP over MONDO but I feel that is better tackled with another rewrite of how the search framework is handled.

Basically so many things go through search now that I'd like to rework it to better display results - and incorporate the things you said, but it's more of a code restructure than a simple change at the end.

The counting did get screwed up if there were no gene relationships (result of my super quick fix for gene symbols) - that shold be fixed now.
Note that gene relationships can get double counted (in direct relationship and as a gene relationship) but the number does now reflect how many rows are shown in the relationship section.

**Comment by EmmaTudini** [comment]
Re-testing
- Logged in as non-admin
1. Searched on breast cancer
Expected output: Should not show obsolete terms
Actual output: Passed

2. Went to https://test.shariant.org.au/ontology/term/MONDO_0010466
Expected output: Should show 14 relationships
Actual output: Passed

3. Went to https://test.shariant.org.au/ontology/term/MONDO_0005217
Expected output: Should show 9 relationships
Actual output: Passed
